fix(logger): force Logger to use local timezone when UTC flag is not set (#3168)
awslabs/aws-lambda-powertools-python: docs/core/logger.md

| **level**: `str`           | `INFO`                                | Logging level                                                                                                                        |
| **location**: `str`        | `collect.handler:1`                   | Source code location where statement was executed                                                                                    |
| **message**: `Any`         | `Collecting payment`                  | Unserializable JSON values are casted as `str`                                                                                       |

> | **timestamp**: `str`       | `2021-05-03 10:20:19,650+0000`        | Timestamp with milliseconds, by default uses default lambda timezone (UTC)                                                                         |

Reviewer: ```suggestion
| **timestamp**: `str`       | `2021-05-03 10:20:19,650+0000`        | Timestamp with milliseconds, by default uses default AWS Lambda timezone (UTC)                                                                         |
```

    --8<-- "examples/logger/src/reordering_log_keys_output.json"
    ```

#### Setting timestamp to custom Timezone

> By default, this Logger and standard logging library emits records using default lambda time(UTC) timestamp. You can enforce this behavior via `utc` parameter.

Reviewer: ```suggestion
By default, this Logger and the standard logging library emit records with the default AWS Lambda timestamp in UTC.
```

I don't think we need to mention the enforce, this is the default behavior. To be honest, I think there is no use for the `utc` field anymore.

> However, if you want to use your preferred timezone to format your log instead, you can use the timezone environment variable `TZ`: Either set it as Lambda Environment Variable. Or setup this value directly in your Lambda Hander.

Reviewer: ```suggestion
However, if you want to use your preferred timezone to format your log instead, you can use the timezone environment variable `TZ`: Either set it as AWS Lambda environment variable, or setup this value directly in your Lambda function.
```

???+ tip

>     `TZ` environment variable will be ingnored if `utc` is set to `True`

Reviewer: ```suggestion
    `TZ` environment variable will be ignored if `utc` is set to `True`
```

=== "setting_custom_timezone.py"

    ```python hl_lines="9 12"
    --8<-- "examples/logger/src/setting_utc_timestamp.py"
    ```

>     1.  if you set TZ in lambda runtime, time.tzset() need to be called. You don't need it when setting TZ in lambda environment variable

Reviewer: ```suggestion
    1.  if you set TZ in your Lambda function, `time.tzset()` need to be called. You don't need it when setting TZ in AWS Lambda environment variable
```
